app/src/main/java/com/android/solvit/provider/ui/request/ListRequestsFeed.kt

```diff
@@ -0,0 +1,434 @@
+        }
+        selectedLocation?.let { GetDirectionsBubble(location = it) { selectedLocation = null } }
+      }
+}
+
+@Composable
+fun InteractionBar(text: String, icon: Int, onClick: () -> Unit = {}) {
+  Column(horizontalAlignment = Alignment.CenterHorizontally) {
+    Text(text = text, fontSize = 14.sp, color = Color.White)
+    IconButton(onClick = { onClick() }) {
+      Image(
+          painter = painterResource(id = icon),
+          contentDescription = text,
+          colorFilter = ColorFilter.tint(Color.White),
+          modifier = Modifier.size(21.dp))
+    }
+  }
+}
+
+@Composable
+fun FilterBar(
+    selectedService: String,
+    selectedFilters: Set<String>,
+    onSelectedService: (String) -> Unit,
+    onFilterChange: (String, Boolean) -> Unit
+) {
+  val filters = listOf("Service", "Near Me", "Due Time")
+
+  LazyRow(
+      modifier = Modifier.testTag("FilterBar"),
+      horizontalArrangement = Arrangement.spacedBy(5.dp),
+      verticalAlignment = Alignment.Top) {
+        items(filters.size) { idx ->
+          val filter = filters[idx]
+          if (filter == "Service") {
+            ServiceChip(
+                selectedService = selectedService,
+                onServiceSelected = { service ->
+                  onSelectedService(service)
+                  onFilterChange("Service", true)
+                })
+          } else {
+            FilterChip(
+                label = filter,
+                isSelected = selectedFilters.contains(filter),
+                onSelected = { isSelected -> onFilterChange(filter, isSelected) })
+          }
+        }
+      }
+}
+
+@Composable
+fun ServiceChip(selectedService: String, onServiceSelected: (String) -> Unit) {
+  var selectedText by remember { mutableStateOf(selectedService) }
+  var showDropdown by remember { mutableStateOf(false) }
+
+  val backgroundColor = if (selectedText != "Service") Color(0xFFFFFAF5) else Color(0xFFFFFFFF)
+  val borderTextColor = if (selectedText != "Service") Color(0xFFFFCD55) else Color(0xFF00C853)
+
+  Box(
+      modifier =
+          Modifier.testTag("ServiceChip").chipModifier(backgroundColor, borderTextColor).clickable {
+            showDropdown = !showDropdown
+          },
+      contentAlignment = Alignment.Center) {
+        Text(
+            text = selectedText,
+            fontSize = 16.sp,
+            lineHeight = 34.sp,
+            fontWeight = FontWeight(400),
+            color = borderTextColor)
+      }
+
+  DropdownMenu(
+      expanded = showDropdown,
+      onDismissRequest = { showDropdown = false },
+      modifier =
+          Modifier.border(1.dp, Color.Gray, RoundedCornerShape(12.dp))
+              .background(Color.White, RoundedCornerShape(12.dp))
+              .testTag("ServiceDropdown")) {
+        Services.entries.forEach { service ->
+          val serviceName =
+              service.name.replace("_", " ").lowercase().replaceFirstChar { it.uppercase() }
```

Comment:
Nit: This line is used twice and could be turned into a function (see 178 to 181)

Reply:
Yes!